aea/aea.py

```diff
@@ -81,7 +79,7 @@ def __init__(self, name: str,
                                      self.decision_maker.ownership_state,
                                      self.decision_maker.preferences,
                                      self.decision_maker.is_ready_to_pursuit_goals)
-        self._resources = None  # type: Optional[Resources]
+        self._resources = resources  # type: Optional[Resources]
         self._filter = None  # type: Optional[Filter]
```

Comment:
The filter can now be initialised here: `Filter(self.resources, self.decision_maker.message_out_queue)`

Reply:
You might also have to check the asserts in the `Filter` class